atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/HikariCPConnectionManager.java

```diff
@@ -88,7 +91,25 @@ private static class State {
     public HikariCPConnectionManager(ConnectionConfig connConfig) {
         this.connConfig = Preconditions.checkNotNull(connConfig, "ConnectionConfig must not be null");
         this.hikariConfig = connConfig.getHikariConfig();
-        hikariConfig.setPoolName(hikariConfig.getPoolName() + "-" + uniquePoolId.incrementAndGet());
+        this.exec = makeExecutorIfNecessary();
+    }
+
+    /**
+     * Manage our own executor so that hikari does not manage it for us. This is
+     * a workaround for the executor leaking in hikari if there is a metric
+     * conflict.
+     */
+    private ScheduledExecutorService makeExecutorIfNecessary() {
+        if (hikariConfig.getScheduledExecutor() != null) {
+            return hikariConfig.getScheduledExecutor();
+        }
+
+        ScheduledThreadPoolExecutor pool =
+                PTExecutors.newScheduledThreadPoolExecutor(1, new ThreadPoolExecutor.DiscardPolicy());
+        pool.setExecuteExistingDelayedTasksAfterShutdownPolicy(false);
```

Comment:
checked this part against https://github.com/brettwooldridge/HikariCP/blob/ed2da5f1f4ef19f871fac12effc0b199706905dc/src/main/java/com/zaxxer/hikari/pool/HikariPool.java#L634